pkg/kubectl/describe.go

```diff
@@ -508,7 +507,7 @@ func describePod(pod *api.Pod, events *api.EventList) (string, error) {
 			}
 		}
 		describeVolumes(pod.Spec.Volumes, out, "")
-		fmt.Fprintf(out, "QoS Class:\t%s\n", qos.InternalGetPodQOS(pod))
+		fmt.Fprintf(out, "QoS Class:\t%s\n", pod.Status.QOSClass)
```

Comment:
looked at this more, i think we can set qosClass in PodStatus in pod strategy PrepareForCreate similar to what we do to set the Pending phase